src/redux/sagas/authSaga.ts

```diff
@@ -13,7 +13,17 @@ import {
 } from "../actions/authActions";
 import AuthTypes from "../types/authTypes";
 
-function postLogin(body: object): Promise<boolean> {
+interface IPostSignUpBody {
+  csrfToken: string;
+  callbackUrl: string | string[];
```

Comment:
`callbackUrl` should just be a string. We can't redirect to multiple pages.

```suggestion
  callbackUrl: string;
```

Reply:
`context.query.callbackUrl` returns `string | string[] | undefined`. Based on your existing logic in `getServerSideProps`, callbackUrl can be a string array.
<img width="300" alt="image" src="https://user-images.githubusercontent.com/34189743/201545153-37c30200-e43a-4e58-b2f2-edca66e4868f.png">

Reply:
And that is why I have `!Array.isArray(callbackUrl)`

<img width="623" alt="Screenshot 2022-11-13 at 4 29 41 PM" src="https://user-images.githubusercontent.com/16737086/201545497-36abdb61-bf24-4dd6-9e5c-5f253f47c91f.png">

Reply:
Updated